editor/components/writing-flow/index.js

```diff
@@ -205,6 +226,31 @@ class WritingFlow extends Component {
 		}
 	}
 
+	componentDidUpdate( prevProps ) {
+		// When selecting a new block, we focus its first editable or the container
+		if (
+			this.props.selectedBlock &&
+			( ! prevProps.selectedBlock || this.props.selectedBlock.uid !== prevProps.selectedBlock.uid )
+		) {
+			const blockContainer = this.container.querySelector( `[data-block="${ this.props.selectedBlock.uid }"]` );
+			if ( ! blockContainer.contains( document.activeElement ) ) {
+				const target = this.getInnerTabbable( blockContainer, this.props.initialPosition === -1 );
+				target.focus();
+				if ( this.props.initialPosition === -1 ) {
+					// Special casing richtext because the two functions at the bo bottomare not working as expected.
```

Comment:
1. Something broke in this comment. :)
2. I think this branching is partly to blame for some broken behavior when multi-selecting with the keyboard (selection starts at the wrong block, off by one, and often this can be avoided by navigating left and right within a block's RichText).
3. On the other hand, we get #3999 fixed for free. :)

Reply:
I updated with a better comment explaining why we need this special case. in https://github.com/WordPress/gutenberg/pull/4872/commits/27f81315a229e48b2924303a0a77e717dd8a07a6

I'm not sure I understand your point `2` here. Are you saying we can avoid this special-case somehow?